packages/lib/services/e2ee/crypto.ts

```diff
@@ -0,0 +1,40 @@
+import { Crypto } from './types';
+import { promisify } from 'util';
+import {
+	getCiphers as nodeGetCiphers, getHashes as nodeGetHashes,
+	randomBytes as nodeRandomBytes,
+	pbkdf2 as nodePbkdf2,
+} from 'crypto';
+
+const crypto: Crypto = {
+
+	getCiphers: (): string[] => {
+		return nodeGetCiphers();
+	},
+
+	getHashes: (): string[] => {
+		return nodeGetHashes();
+	},
+
+	randomBytes: async (size: number): Promise<Buffer> => {
+		const randomBytesAsync = promisify(nodeRandomBytes);
+		return randomBytesAsync(size);
+	},
+
+	pbkdf2Raw: async (password: string, salt: Buffer, iterations: number, keylen: number, digest: string): Promise<Buffer> => {
+		const digestMap: { [key: string]: string } = {
+			'sha-1': 'sha1',
+			'sha-224': 'sha224',
+			'sha-256': 'sha256',
+			'sha-384': 'sha384',
+			'sha-512': 'sha512',
+			'ripemd-160': 'ripemd160',
+		};
+		const digestAlgorithm: string = digestMap[digest.toLowerCase()] || digest;
```

Comment:
I don't really get this part - are the digestMap the list of supported digests? In that case shouldn't that be exposed to the library user? In other words, it seems there should be an enum that tells the list of supported digests.

And `digest` should probably have this type.

Also please don't use `toLowerCase()` for this - the input digest should be strictly correct and checked at compile time using a type.

Reply:
The `digestMap` of `pbkdf2Raw()` is for mapping the name according to the implementations. For `Node.js`, the hash algorithms are named like `sha1`, `sha256` while for `react-native-quick-crypto` they are named like `SHA-1`, `SHA-256`. I planned to use the string type for the name, just like how `Node.js` does. But now I will change it to a enum because `react-native-quick-crypto` only supports a limited set of them.